torchgeo/datasets/soda.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SODA datasets."""
+
+import json
+import os
+from collections.abc import Callable
+from typing import ClassVar
+
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from shapely.geometry import MultiPoint, Polygon
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, check_integrity, download_and_extract_archive, download_url
+
+
+class SODAA(NonGeoDataset):
+    """SODA-A dataset.
+
+    The `SODA-A <https://shaunyuan22.github.io/SODA/>`_ dataset is a high resolution aerial imagery dataset for small object detection.
```

Comment:
What editor are you using? There should be a way to add a line at 88 chars so you know where to add a new line...